momepy/preprocessing.py

```diff
@@ -248,10 +248,16 @@ def remove_false_nodes(gdf):
             lines=False,
         )
         loops = combined[combined.is_ring]
+
         node_ix, loop_ix = loops.sindex.query(nodes.geometry, predicate="intersects")
         for ix in np.unique(loop_ix):
             loop_geom = loops.geometry.iloc[ix]
             target_nodes = nodes.geometry.iloc[node_ix[loop_ix == ix]]
+            if target_nodes.has_z.any():
+                raise ValueError(
+                    "One or more input nodes is 3D. "
+                    "Force 2D with ``geopandas.GeoSeries.force_2d()``."
+                )
             if len(target_nodes) == 2:
                 node_coords = shapely.get_coordinates(target_nodes)
                 coords = np.array(loop_geom.coords)
```

Comment:
```suggestion
            if len(target_nodes) == 2:
                node_coords = shapely.get_coordinates(target_nodes)
                coords = shapely.get_coordinates(loop_geom)
```

Another option is this, which makes the whole thing work. We just drop Z coordinates in that case.

Reply:
So do we think:
* [ ] a check + warning if 3D coords present
* [ ] simply forcing 2D without check

Reply:
I would make a note in the docstring and let it use 2d.